lib/primitives/mtx.js

```diff
@@ -1888,6 +1888,35 @@ class CoinSelector {
     this.fund();
   }
 
+  /**
+   * Initiate selection using Single Random Draw selection.
+   * @param {Number} target - Selection target
+   * @returns {Number[]} selected - array of indicies of selected coins
+   */
+
+  selectSRD(target) {
+    const coins = this.coins.map((coin, index) => {
+      coin.pos = index;
+      return coin;
+    });
+    coins.sort(() => Math.random() - 0.5);
```

Comment:
is this faster than keeping `this.coins` as is, and picking random indexes? Maybe using `splice()` to pull selected coins out?